mitosheet/mitosheet/tests/saved_analyses/test_save_utils.py

```diff
@@ -329,4 +329,37 @@ def test_save_replays_overwrite_by_ids_propererly():
 
     assert new_mito.dfs[1].equals(
         pd.DataFrame({'A': [1, 2, 3], 'A count': [1, 1, 1]})
-    )
\ No newline at end of file
+    )
+
+def test_save_and_replay_different_interface_version_works():
+    mito = create_mito_wrapper([1, 2, 3])
+    mito.mito_backend.steps_manager.public_interface_version = 100
+
+    random_name = 'UUID-test_save' + str(random.random())
+    mito.save_analysis(random_name)
+
+    saved_analysis = read_and_upgrade_analysis(random_name)
+    assert saved_analysis is not None
+    assert len(saved_analysis['steps_data']) == 0
+    assert saved_analysis['public_interface_version'] == 100
+    print(saved_analysis)
+
+    new_mito = create_mito_wrapper([1, 2, 3])
+    new_mito.replay_analysis(random_name)
+
+    assert new_mito.mito_backend.steps_manager.public_interface_version == 100
+
+def test_replay_failed_analysis_does_not_change_public_interface_version():
+    mito = create_mito_wrapper([1, 2, 3])
+    mito.delete_columns(0, ['A'])
+    mito.mito_backend.steps_manager.public_interface_version = 100
```

Comment:
Want to make starting_val = 100 here and then use it in line 355 to make changing it easy

Reply:
I don't see a reason we'd change it, so I think it's cool!